Application/EdFi.Ods.SandboxAdmin/Controllers/Api/ClientController.cs

```diff
@@ -92,17 +93,18 @@ public async Task<IActionResult> GetClients()
             {
                 var currentException = e;
                 var counter = 0;
-                var message = string.Empty;
+                var message = new StringBuilder();
 
                 while (currentException != null)
                 {
                     Trace.TraceError("{0}: {1}", currentException.Message, currentException.StackTrace);
-                    message += string.Format("{0}:{1}", counter, currentException.Message);
+                    message.AppendLine($"{counter}:{currentException.Message}");
                     counter++;
                     currentException = currentException.InnerException;
                 }
 
-                throw new Exception(string.Format("Unhandled exception with these messages: {0}", message));
+                // This is not an ideal exception type, but it is better than a plain Exception
+                throw new InvalidOperationException(string.Format("Unhandled exception with these messages: {0}", message));
```

Comment:
This is where I disagree with the advice, and we've had this discussion previously, but to recap my opinion on this. I feel that there are only 2 times when a specifically typed exception is thrown in an _application_ (as opposed to a reusable library/framework):
- When there is an existing exception that matches perfectly as documented. `InvalidOperationException` is documented as "the exception that is thrown when a method call is invalid for the object's current state.". That doesn't fit here. There is nothing about the state of the controller that impacted the ability to invoke the `GetClients` method (controllers don't even have state, generally).
- When there is a need to attach definitive semantics to an exception so that it can be handled in a particular way elsewhere in the system (e.g. to be able to make a decision on which status code to return in the HTTP response). In most cases this is not needed (a 500 status is perfect) and the addition of new custom exception types for each occurrence is an onerous development and maintenance requirement, while an arbitrary usage of a _different_ type of the exception than `Exception` is pointless and equally lacking in semantic meaning.

Reply:
I actually agree with that position, though I had to jog my memory on those details by actually working through the example.

The only reason I would consider leaving this as a Suggestion (which, to be honest, might not ever be noticed) rather than completely disabling the advice is to help prompt people to _think_ about whether or not there is an existing appropriate exception.